src/modules/scraper/adapter/messaging/SpeedUpEventsConsumer.ts

```diff
@@ -0,0 +1,65 @@
+import { OnQueueFailed, Process, Processor } from "@nestjs/bull";
+import { Logger } from "@nestjs/common";
+import { Job } from "bull";
+import { DepositFilledDateQueueMessage, ScraperQueue, SpeedUpEventsQueueMessage } from ".";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Deposit } from "../../model/deposit.entity";
+import { Repository } from "typeorm";
+import { ScraperQueuesService } from "../../service/ScraperQueuesService";
+
+@Processor(ScraperQueue.SpeedUpEvents)
+export class SpeedUpEventsConsumer {
+  private logger = new Logger(SpeedUpEventsConsumer.name);
+
+  constructor(
+    @InjectRepository(Deposit) private depositRepository: Repository<Deposit>,
+    private scraperQueuesService: ScraperQueuesService,
+  ) {}
+
+  @Process()
+  private async process(job: Job<SpeedUpEventsQueueMessage>) {
+    const { depositId, depositSourceChainId } = job.data;
+    const deposit = await this.depositRepository.findOne({ where: { sourceChainId: depositSourceChainId, depositId } });
+
+    if (!deposit) {
+      throw new Error(
+        `RequestedSpeedUpDeposit event for deposit with depositId '${depositId}' and sourceChainId '${depositSourceChainId}' could not be processed: deposit not found in db`,
+      );
+    }
+
+    if (this.isSpeedUpAlreadyProcessed(deposit, job.data)) {
+      this.logger.warn("RequestedSpeedUpDeposit event already processed");
+      return;
+    }
+
+    await this.processSpeedUpEventQueueMessage(deposit, job.data);
+
+    this.scraperQueuesService.publishMessage<DepositFilledDateQueueMessage>(ScraperQueue.DepositFilledDate, {
+      depositId: deposit.id,
+    });
+  }
+
+  public async processSpeedUpEventQueueMessage(deposit: Deposit, data: SpeedUpEventsQueueMessage) {
+    const { transactionHash, newRelayerFeePct, blockNumber, depositSourceChainId } = data;
+
+    deposit.speedUps = [
+      ...deposit.speedUps,
+      { hash: transactionHash, newRelayerFeePct, blockNumber, depositSourceChainId },
+    ].sort((a, b) => b.blockNumber - a.blockNumber);
+
+    deposit.depositRelayerFeePct = deposit.speedUps[0].newRelayerFeePct;
+
+    return this.depositRepository.save(deposit);
```

Comment:
Can we use `.update` instead of `.save()`? I will update `FillEventsConsumer` to use the same approach

Reply:
done